src/nvim/option.c

```diff
@@ -4331,6 +4330,10 @@ set_num_option (
   if ((opt_flags & (OPT_LOCAL | OPT_GLOBAL)) == 0)
     *(long *)get_varp_scope(&(options[opt_idx]), OPT_GLOBAL) = *pp;
 
+  if (!curbuf->terminal) {
+    curbuf->b_p_scbk = -1;
```

Comment:
When, say, `:set scrollback=99` is called, `set_num_option` will have `pp == &curbuf->b_p_scbk`, i.e., it validates the local value and above here it copies the local to the global value. So we need to reset the value to `-1` after it's been copied to the global value. This change seemed the least invasive to handle this case.